std/array.d

```diff
@@ -451,17 +451,20 @@ unittest
 }
 
 ///
-@safe pure nothrow unittest
+@safe @nogc pure nothrow unittest
 {
+    import std.algorithm : equal;
+    import std.range : only;
+    
     auto arr = [1,2,3,4].staticArray;         // no need to declare size
     static assert(is(typeof(arr) == int[4])); // i is a static array
-    assert(arr == [1,2,3,4]);
+    assert(arr[].equal(only(1,2,3,4)));
```

Comment:
Yes, I assumed it wouldn't compile. Restored the old assert. I think it compiles because the literal is interpreted as a static array probably because arr is one.